File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        sentinel = ListNode()
        head = sentinel
        while list1 and list2:
            if list1.val < list2.val:
                sentinel.next = list1
                list1 = list1.next
            else:
                sentinel.next = list2
                list2 = list2.next
            sentinel = sentinel.next
        if list1:
            sentinel.next = list1
        else:
            sentinel.next = list2
        return head.next

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode slow = new ListNode(0, head);
        ListNode sentinel = slow;
        ListNode fast = head;
        int split = 0;
        while (fast != null && split < n) {
            fast = fast.next;
            split++;
        }
        while (fast != null) {
            fast = fast.next;
            slow = slow.next;
        }
        slow.next = slow.next.next;
        return sentinel.next;
    }
}
